File: Lista4/Questao4/Questao4/src/questao4/ContaCorrente.java
package questao4;

public class ContaCorrente {
    private float saldo;
    private float creditoLimite;
    private int pontuacaoSerasa;

    public ContaCorrente(){
        saldo = 0;
        creditoLimite = 0;
    }
    
    public void sacar(float valor) throws LimiteAtingido{
        if(valor > saldo+creditoLimite)
            throw new LimiteAtingido("Limite de credito atingido!");
        saldo -= valor;
    }
    
    public void depositar(float valor) throws DepositoInvalido{
        if(valor < 1)
            throw new DepositoInvalido(valor);
        saldo += valor;
    }
    
    public void setValorLimite(float valor) throws LimiteInvalido{
        creditoLimite = valor;
        
    }
    
    public float retornarSaldo(){
        return saldo;
    }
    
    public float retornarLimite(){
        return creditoLimite;
    }
}

File: Lista12/ContaBancaria/ContaCorrente.cpp
#include "ContaCorrente.h"
#include <iostream>
conta::conta() {
    saldo = 0;
}

conta::~conta() {
}

int conta::getSaldo(){
    return saldo;
}

void conta::depositar(int valor) {

    if (valor <= 0)
        cout << "Saldo inválido!";

    else{
        
        int novoSaldo = saldo + valor;
        saldo = novoSaldo;
        
    }

}

void conta::sacar(int valor){
    
    if(valor > saldo)
        cout << "saldo insuficiente!";
    
    else{
        
        int novoSaldo = saldo - valor;
        saldo = novoSaldo;
        
    }
}